test/upload-file-test.js

```diff
@@ -188,54 +190,106 @@ suite.addBatch({
                                 topic: function(doc, feed, pair, cl) {
                                     var browser = new Browser(),
                                         callback = this.callback,
                                         user = pair.user;
 
-                                    browser.visit("http://localhost:4815/main/login")
-                                        .then(function() {
+                                    Step(
+                                        function() {
+                                            browser.visit("http://localhost:4815/main/login", this);
+                                        },
+                                        function(err) {
+                                            if (err) throw err;
+
                                             browser.fill("#nickname", user.nickname)
                                                 .fill("#password", "stormtroopers_hittin_the_ground")
-                                                .pressButton("div#loginpage form button[type=\"submit\"]", function() {
-                                                    callback(null, browser);
-                                                }, callback);
-                                        }, callback);
+                                                .pressButton("div#loginpage form button[type=\"submit\"]", this);
+                                        },
+                                        function(err) {
+                                            callback(err, browser);
+                                        }
+                                    );
                                 },
                                 "it works": function(err, br) {
                                     assert.ifError(err);
                                     br.assert.success();
                                 },
                                 "and we get the file from the web interface while logged in": {
-                                    topic: function(br, doc, feed, pair, cl) {
-                                        var cred = makeCred(cl, pair),
-                                            browser = br,
+                                    topic: function(br, doc) {
+                                        var browser = br,
                                             callback = this.callback,
                                             url = doc.fullImage.url;
 
                                         browser.visit(url, function(err) {
-                                            callback(err, browser);
+                                            // when is false sends a new param for next test
+                                            // and don't use br.assert because send a browser
+                                            // as the new parameter for next test
+                                            callback(err || browser.status !== 200 || null);
```

Comment:
I'm not sure I understand why this change was made; can you clarify further? Why isn't passing `browser` a good idea?

Reply:
because you will have duplicate `browser` instance  `topic: function(br, br, doc)`, and zombie for some reason fail when the response 403 and 402 happened in the same `topic: function..`

Reply:
Gotcha. Typo: `we'll have a` instead of `we'll a`